pypsa-to-iamc

```diff
@@ -0,0 +1,634 @@
+                             "Winter_peak_consumpt") :
+              sh=sheets[sheet]
+            
+              #Hourly power consumption : Buildings, Industry, Transportation  
+            
+              var[sh][sheet_var[sheet]+'|Electricity'] = 0.001*ct*gen_des(time((n.loads_t.p[[i for i in n.loads.index if i=='FR'+'0 0']]).squeeze()))
+              var[sh][sheet_var[sheet]+'|Buildings|Heating'] = 0.001*ct*gen_des(time(n.loads_t.p.filter(like ='heat').filter(like=country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Buildings|Residential|Heating'] = 0.001*ct*gen_des(time(n.loads_t.p.filter(like ='heat').filter(like='residential').filter(like=country).sum(axis=1)))
+              var[sh][sheet_var[sheet]+'|Buildings|Services|Heating'] = 0.001*ct*gen_des(time(n.loads_t.p.filter(like='heat').filter(like ='services').filter(like=country).sum(axis=1)))
+            
+              var[sh][sheet_var[sheet]+'|Industries|Electricity'] = 0.001*ct*gen_des(time(n.loads_t.p.filter(like ='industry electricity').filter(like =country).sum(axis=1).squeeze()))
+              var[sh][sheet_var[sheet]+'|Transportation'] = (0.001*ct*(gen_des(time(n.loads_t.p.filter(like ='transport').filter(like =country).sum(axis=1)))
+                +gen_des(time(n.loads_t.p.filter(like ='shipping').filter(like =country).sum(axis=1)))))
+                                      
+            ### Efficiency
+            sh=sheets["Efficiency_supply"]  
+            ct=3.6e-3    #convert MWh to TJ
+                  
+            
+            #Efficiency : Nuclear,Coal (Lignite),Natural gas(OCGT, CCGT, CHP, CHP CC),Biomass (CCS),Hydrogen,Oil  
+            
+            for v in var[sheets['fuel_consumption_supply']].keys() :  #For values where output power and fuel is already calculated
+                var[sh][v.replace('Fuel consumption','Efficiency')]=ct*var[sheets['Yearly_generation_supply']][v.replace('Fuel consumption','Generation|Yearly')]\
+                                                                        /var[sheets['fuel_consumption_supply']][v]
+            
+             #Efficiency : hydro (reservoir, ror)
+
+            var[sh]['Efficiency|Electricity|Hydro|river'] = n.generators.efficiency.filter(like ='ror').filter(like =country).mean()
+            var[sh]['Efficiency|Electricity|Hydro|dam'] = n.storage_units.efficiency_dispatch.filter(like ='hydro').filter(like =country).mean()
+            var[sh]['Efficiency|Electricity|Hydro'] = var[sh]['Efficiency|Electricity|Hydro|dam']   #currently equal in the model since eff(dam)=eff(river)
+             
+                
+             #Efficiency : storage (PHS, battery, H2 storage)   
+            
+            var[sh]['Efficiency|Flexibility|Electricity Storage|Medium duration'] = (
+                  (n.storage_units.efficiency_dispatch.filter(like ='PHS').filter(like =country).mean()*n.storage_units.efficiency_store.filter(like ='PHS').filter(like =country).mean()+
```

Comment:
I don't think this is correct. 
Let's check if we need to report round-trip efficiency (charging_efficiency*discharging_efficiency)
The current calculation discharging_efficiency*¿self-discharge efficiency? does not make sense

Reply:
It's basically means if you want to store electricity in let's say batteries, the overall efficiency of the power you give to the battery and the power you receive from it is the roundtrip efficiency.